src/control/positionControl/CollisionDetector.cpp

```diff
@@ -9,7 +9,7 @@ namespace rtt::ai::control {
 bool CollisionDetector::isCollisionBetweenPoints(const Vector2& initialPoint, const Vector2& nextPoint) {
     bool isFieldColliding = field ? !isPointInsideField(nextPoint) || isPointInDefenseArea(nextPoint) : false;
 
```

Comment:
As I understood, the keeper will have a different behavior. You can always just use the functions individually.

> Returning isFieldColliding when it is true, might give a performance boost since getRobotCollisionBetweenPoints doesn't have to be called anymore

Fair, I'll change the order